core/src/main/kotlin/org/rewedigital/katana/dsl/Dsl.kt

```diff
@@ -83,11 +83,11 @@ inline fun <reified T> ModuleBindingContext.eagerSingleton(
  * Declares an alias of type [A] with optional [name] of an already existing binding of type [T] with optional name
  * [originalName]. An alias is accomplished by and nothing else than a simple [factory] binding.
  *
- * A binding like `alias<MyClass, MyInterface>()` should be conceived as "alias MyClass as MyInterface".
+ * A binding like `alias<MyInterface, MyClass>()` should be conceived as "create alias MyInterface for MyClass".
  */
-inline fun <reified T : A, reified A> ModuleBindingContext.alias(
-    originalName: String? = null,
-    name: String? = null
+inline fun <reified A, reified T : A> ModuleBindingContext.alias(
+    name: String? = null,
```

Comment:
If one wants to name the alias, do we really need both reified types? Wouldn't it be enough to specify the type be taken from since anyways the binding won't listen to the type.

Reply:
I mean, maybe declaring a separate function:
```
inline fun <reified A, reified T : A> ModuleBindingContext.alias(
    originalName: String? = null
) {
    factory<A>(name = name) { get<T>(name = originalName) }
}

inline fun <reified T> ModuleBindingContext.alias(
    name: String? = null,
    originalName: String? = null
) {
    factory(name = name) { get<T>(name = originalName) }
}
```

Reply:
But then we would also require a third `alias` where the user can rename **and** change the type. I guess the most common use case for `alias` is to bind to another super type. So if we leave the function as is, renaming is optional but also possible with a little bit of boilerplate code.

Reply:
Renaming and changing the type really makes sense? Katana doesn't seem to look at types for named instances...

Reply:
If you agree that aliasing one type for another is the main use case I would like to keep the function as is for now ;)

Reply:
Agreed.